src/modules/jcms_admin/src/Plugin/views/field/ContentLink.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Drupal\jcms_admin\Plugin\views\field;
+
+use Drupal\Core\Database\Database;
+use Drupal\node\NodeInterface;
+use Drupal\views\Plugin\views\field\FieldPluginBase;
+use Drupal\views\ResultRow;
+
+/**
+ * Field handler to present the path to the content on journal.
+ *
+ * @ingroup views_field_handlers
+ *
+ * @ViewsField("jcms_content_link")
+ */
+class ContentLink extends FieldPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function query() {
+    // Leave empty to avoid a query on this field.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render(ResultRow $values) {
+    /** @var \Drupal\node\NodeInterface $content */
+    $content = $values->_entity;
+    $journal = \Drupal::getContainer()->hasParameter('journal') ? \Drupal::getContainer()->getParameter('journal') : [];
+    $path = $this->getContentPath($content);
+    if (!$path || !$journal || !isset($journal['path'])) {
+      return '';
+    }
+    else {
+      $base_path = $journal['path'];
+      $render = [
+        [
+          '#theme' => 'clipboard_simple',
+          '#text' => $base_path . $path,
+        ],
+      ];
+      if ($preview_path = $journal['preview']) {
+        $render[] = [
+          '#theme' => 'clipboard_simple',
+          '#text' => $preview_path . $path,
+          '#button_text' => 'Copy preview url',
+        ];
+      }
+      return $render;
+    }
+  }
+
+  /**
+   * Get content path.
+   */
+  private function getContentPath(NodeInterface $content) {
+    $whitelist = [
+      'annual_report',
+      'blog_article',
+      'collection',
+      'event',
+      'interview',
+      'labs_experiment',
+      'podcast_episode',
+    ];
+    $sns = \Drupal::service('jcms_notifications.entity_crud_notification_service');
+    /** @var \Drupal\jcms_notifications\Notification\BusOutgoingMessage $message */
+    if (in_array($content->bundle(), $whitelist) && $message = $sns->getMessageFromEntity($content)) {
+      $stem = $message->getTopic();
+      $id = $message->getId();
+      $slug = NULL;
+      $seperator = '/';
+      switch ($content->bundle()) {
+        case 'annual_report':
+          $id = NULL;
+          break;
+
+        case 'blog_article':
+          $stem = 'inside-elife';
+          $slug = \Drupal::service('slugify')->slugify($content->getTitle());
```

Comment:
not sure if the slugify algorithm here will always be 100% identical to what `journal` does. However, `journal` will accept also incorrect slugs redirecting them to the canonical URL. So this is robust :+1: